File: Array/maxProfit/maxProfit.c
int maxProfit(int *prices, int pricesSize)
{
    int max = 0;
    int minprice = 1e9;
    int profit = 0;
    for (int i = 0; i < pricesSize; i++)
    {
        profit = prices[i] - minprice;
        if (profit > max)
        {
            max = profit;
        }
        if (minprice > prices[i])
        {
            minprice = prices[i];
        }
    }
    return max;
}

File: Array/maxProfit/maxProfit.go
// https://leetcode.cn/problems/best-time-to-buy-and-sell-stock/description/
package main

import "math"

func maxProfit(prices []int) int {
	max := 0
	minprice := math.MaxInt
	profit := 0
	for i := 0; i < len(prices); i++ {
		profit = prices[i] - minprice
		if profit > max {
			max = profit
		}
		if minprice > prices[i] {
			minprice = prices[i]
		}
	}
	return max
}

func main() {

}
